core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -91,7 +90,12 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
   /** List all the files currently stored on disk by the disk manager. */
   def getAllFiles(): Seq[File] = {
     // Get all the files inside the array of array of directories
-    subDirs.flatten.filter(_ != null).flatMap { dir =>
+    subDirs.flatMap { dir =>
```

Comment:
The earlier change makes sense, given what the intent of the code seems to be. Is this one needed? cloning the array doesn't change any guarantees about whether a call to this `get` method sees a particular update above or not. The cloning of the array, like the read, will happen before or after any given update. Or am I missing some correctness corner-case here that means something throws an exception or something? just hoping to avoid complicating this.